activerecord/test/cases/counter_cache_test.rb

```diff
@@ -30,6 +31,23 @@ class ::SpecialReply < ::Reply
     belongs_to :special_topic, foreign_key: "parent_id", counter_cache: "replies_count"
   end
 
+  class ::ReplyWithTouch < ::Reply
+    belongs_to :topic_with_after_commit_callback, foreign_key: "parent_id", counter_cache: "replies_count", touch: true
+
+    # after_commit :explicitly_touch_topic
+    # def explicitly_touch_topic
+    #   topic_with_after_commit_callback.touch
+    # end
+  end
+
+  class ::TopicWithAfterCommitCallback < ::Topic
+    after_commit :increase_task_count
```

Comment:
This `after_commit` should be executed after `ReplyWithTouch` is created (`TopicWithAfterCommitCallback` has a counter_cache column: `replies_count` that gets updated correctly, but the `after_commit` doesn't get called)